test/run-all-scenarios.sh

```
@@ -0,0 +1,635 @@
+
+        if [[ -z "${kubeconfig_secret}" ]]; then
+            log_warning "Kubeconfig secret not found (cluster may not be fully ready)"
+        else
+            log_success "Kubeconfig secret found: ${kubeconfig_secret}"
+        fi
+    fi
+
+    # Verify ACM ManagedCluster
+    log_info "Verifying ACM ManagedCluster auto-import"
+    if [[ "${DRY_RUN}" == "false" ]]; then
+        local managed_cluster=$(kubectl get managedcluster -n "${test_namespace}" 2>/dev/null | wc -l)
```

Shellcheck (0.11.0)

[warning] Declare and assign separately to avoid masking return values.

(SC2155)